### PR TITLE
[Concept Entry] Postgresql subscripts

### PR DESCRIPTION
**Pull request by SaviDahegaonkar**
### Description

New Concept entry on PostgreSQL subscripts

### Issue Solved
#4587 

### Type of Change

- Adding a new entry

### Checklist

- [ ] All writings are my own.
- [ ] My entry follows the Codecademy Docs style guide.
- [ ] My changes generate no new warnings.
- [ ] I have performed a self-review of my own writing and code.
- [ ] I have checked my entry and corrected any misspellings.
- [ ] I have made corresponding changes to the documentation if needed.
- [ ] I have confirmed my changes are not being pushed from my forked `main` branch.
- [ ] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [ ] I have linked any issues that are relevant to this PR in the `Issues Solved` section.

### CODE REVIEW

File: content/postgresql/concepts/subscripts/subscripts.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
+---
+Title: 'Subscripts'
+Description: 'In PostgreSQL, subscripts are used to perform various operations on arrays. They offer a concise way to handle and manage array data.'
```
Comment (PragatiVerma18) [inline content/postgresql/concepts/subscripts/subscripts.md:3]: The definition can be clearer by mentioning the operations that can be performed. Also, the position is the most important aspect of subscripts; we should use it to explain the role of subscripts.

Description can be something like this -

```suggestion
Description: 'Subscripts are used to access, update, or manipulate specific elements of an array by their position or index.'
```

File: content/postgresql/concepts/subscripts/subscripts.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+```pseudo
+array_name[subscript]
+```
+- `array_name`: The name of the array being accessed.
+- `subscript`: The index position of the element to be accessed within the array.
+
+Alternatively, for multi-dimensional array the following syntax is used:
+
+```pseudo
+array_name[subscript_1, subscript_2, ..., subscript_n]
```
Comment (PragatiVerma18) [inline content/postgresql/concepts/subscripts/subscripts.md:34]: In PostgreSQL, multi-dimensional array elements are accessed using nested square brackets, where each set of brackets corresponds to a specific dimension. Refer [docs](https://www.postgresql.org/docs/current/arrays.html) for more info.

The syntax for accessing multi-dimensional arrays in Postgres is wrong, and the correct syntax is as follows -

```suggestion
```pseudo
array_name[subscript_1][subscript_2]...[subscript_n]
```

File: content/postgresql/concepts/subscripts/subscripts.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+```
+- `array_name`: The name of the array being accessed.
+- `subscript_1, subscript_2, ..., subscript_n`: The index positions for each dimension of the array to access the specific element.
+
+## Example
+
+Suppose, we have a table `students` with columnname as `grades` that stores a multi-dimensional array of test scores:
```
Comment (PragatiVerma18) [inline content/postgresql/concepts/subscripts/subscripts.md:41]: This sentence can be a bit more clear as follows -

```suggestion
Suppose we have a table `students` with column name `grades` that stores a multi-dimensional array of test scores:
```

File: content/postgresql/concepts/subscripts/subscripts.md [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+INSERT INTO students (name, grades)
+VALUES 
+    ('Bryan', ARRAY[[98, 60], [75, 52]]);
+```
+If we want to access Bryan's score, we will run the below query:
+
+```sql
+SELECT grades[2, 2] AS second_subject_score
+FROM students
+WHERE name = 'Bryan';
```
Comment (PragatiVerma18) [inline content/postgresql/concepts/subscripts/subscripts.md:62]: In PostgreSQL, for multi-dimensional arrays, multiple subscripts should be used with separate square brackets for each dimension. The correct query should be:

```suggestion
SELECT grades[2][2] AS second_subject_score
FROM students
WHERE name = 'Bryan';
```

**Review by PragatiVerma18 — APPROVED** [review]
Hey @SaviDahegaonkar, this entry looks good to go now! 🚀

**Comment by SaviDahegaonkar** [comment]
Sorry @PragatiVerma18, really was busy with other PR's, so I was unavailable. Next time I will take care of the changes.

Thanks,
Savi

**Pull request merged**